### PR TITLE
Using attributes on fields and properties to exclude from comparison.

### PR DESCRIPTION
**Pull request by aksar9211**
Valerii Tereshchenko, I am using object comparer in my project. I used to list out the ignore member before comparison, but In several cases, there are nested parent-child models that go into the loop recursively and result in StackOverflowException.
Also instead of excluding the member at the comparer site, It is more useful for us to specify it at the property and field. So, I am proposing the above solution for this issue. Please consider adding it to the live project.

Usage:

```
[Comparison(ComparisonStatus.ExcludeFromComparison)]
public ObservableCollection <MyClass> myObj{get; set;}

[Comparison(ComparisonStatus.ExcludeFromComparison)]
public MyClass parent{get; set;}
```

### CODE REVIEW

File: ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+  /// <summary>
+  /// Specifies the status of comparison i.e. whether to include in comparison or not
+  /// </summary>
+  public enum ComparisonStatus
+  {
+    /// <summary>
+    /// Specifies that the element on which it is applied has to be excluded from comparison
+    /// </summary>
+    ExcludeFromComparison
```
Comment (ValeraT1982) [inline ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs]: What is the benefits of having this enum with 1 value?

File: ObjectsComparer/ObjectsComparer/Attributes/ComparisonAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace ObjectsComparer.Attributes
+{
+  using System;
+  using ObjectsComparer.Enums;
+
+  /// <summary>
+  /// Class is used to specify whether the element on which it is applied will have comparison effect
+  /// </summary>
+  [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+  public sealed class ComparisonAttribute : Attribute
```
Comment (ValeraT1982) [inline ObjectsComparer/ObjectsComparer/Attributes/ComparisonAttribute.cs]: Consider renaming to something like IgnoreInComparison

**Review by ValeraT1982 — COMMENTED** [review]
Hi,

Thanks a lot fore the pull request. Can you please address my comments and add tests?

File: ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs (hunk 1 above)
Comment (aksar9211) [reply ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs]: Currently, I have added just one value in the enum as per my requirements, but further, we can extend enums to modify more settings like `IgnoreRecursiveComparison ` - if we want only one item to ignore for recursive comparison. So I am keeping my options open to manage comparison settings in one enum class.

File: ObjectsComparer/ObjectsComparer/Attributes/ComparisonAttribute.cs (hunk 2 above)
Comment (aksar9211) [reply ObjectsComparer/ObjectsComparer/Attributes/ComparisonAttribute.cs]: Ok. That won't be a problem.

**Review by aksar9211 — COMMENTED** [review]
I have added the tests as suggested. Please review again.

File: ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs [hunk 3]
```diff
@@ -12,8 +12,8 @@ namespace ObjectsComparer.Enums
   public enum ComparisonStatus
```
Comment (ValeraT1982) [inline ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs]: I was thinking about removing this enum completely. It simplifies things a lot.

**Review by ValeraT1982 — CHANGES_REQUESTED** [review]

**Comment by aksar9211** [comment]
As per your suggestions, I have removed the ComparisonStatus enum class and renamed the ComparisonAttribute to IgnoreInComparisonAttribute. 
I have also updated the unit testes. 
Please review again.

File: ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs (hunk 3 above)
Comment (aksar9211) [reply ObjectsComparer/ObjectsComparer/Enums/ComparisonStatus.cs]: Ok. I have removed it.

**Review by ValeraT1982 — COMMENTED** [review]
Did you run all unit tests with this change?

**Comment by aksar9211** [comment]
Yes. I did. All the tests passed except one:
File: ComparerGenericEnumerableTests.cs  Line no.:550
Assert.AreEqual(DifferenceTypes.MissedElementInFirstObject, differences.First().DifferenceType);

the difference type is: DifferenceTypes.NumberOfElementsMismatch

This test is failing in the original project as well.

**Comment by ValeraT1982** [comment]
Thanks for your contribution

**Comment by aksar9211** [comment]
Thank you for merging the code.
How to get the nuget package for the same?

**Comment by ValeraT1982** [comment]
Currently the only option is to download the source code. I'm going to release next version within couple of month.

**Comment by swinslade** [comment]
Hi, I downloaded and built the source back in June 2022 just so I could use the [IgnoreInComparison] attribute (which is working perfectly in my project), is there any chance of a new release (nuget package) for these changes in the near future? 

Thanks for the great work on this project!

**Comment by ValeraT1982** [comment]
Still working on it. Sorry. A lot of work.

**Pull request merged**